corehq/apps/data_dictionary/static/data_dictionary/js/data_dictionary.js

```diff
@@ -0,0 +1,61 @@
+(function ($, _) {
+    var CaseProperty = function (data) {
+        this.name = ko.observable(data.name);
+        this.type = ko.observable(data.type);
+        this.description = ko.observable(data.description);
+    };
+
+    var CaseType = function (name) {
+        var self = this;
+        self.name = ko.observable(name);
+        self.properties = ko.observableArray();
+
+        self.init = function (properties) {
+            _.each(properties, function (property) {
+                self.properties.push(new CaseProperty(property));
+            });
+        };
+    };
+
+    var DataDictionaryModel = function (dataUrl) {
+        var self = this;
+        self.caseTypes = ko.observableArray();
+        self.activeCaseType = ko.observable();
+
+        self.init = function () {
+            $.getJSON(dataUrl)
+            .done(function (data) {
+                _.each(data.properties, function (property) {
```

Comment:
a bit weird that `data.properties` is actually case types and `property` is actually a case type